app/workers/asset_manager_update_asset_worker.rb

```diff
@@ -12,10 +12,11 @@ def perform(attachment_data, legacy_url_path, new_attributes = {})
     rescue GdsApi::HTTPNotFound
       asset_missing = true
     end
+    asset_deleted = attributes['deleted']
 
-    if asset_missing && attachment_data.deleted?
+    if (asset_missing || asset_deleted) && attachment_data.deleted?
```

Comment:
I _think_ it's OK to treat both deleted and missing attachments as no-ops as long as the `AttachmentData` is deleted. In both cases (deleted or missing) the response from Asset Manager will be a 404 which is equivalent to the 404 response from Whitehall for a deleted `AttachmentData`.

Reply:
I've changed my mind about this based on your comment. I've now updated `AssetManagerUpdateAssetWorker#perform` so that we no longer rescue `GdsApi::HTTPNotFound` exceptions.